### ISSUE TITLE
False reporting - "No Java runtime present" on MacOSX

### ISSUE THREAD

**Issue by jr0cket** [oakes/Nightcode#194]
A new install of Nightcode 2.1.2 today on MacOSX 10.11.6 (OSX El Capitan) reports that no Java runtime is present, requesting install.  The install was from the .dmg package from the Nightcode website.

This computer has java installed and `java -version` reports
```
java version "1.8.0_102"
Java(TM) SE Runtime Environment (build 1.8.0_102-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.102-b14, mixed mode)
```

The JAVA_HOME variable is also set and points to this installation of Java.  Using `echo $JAVA_HOME` reports

```
/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home
```

We are run a ClojureBridge event in a few weeks, so wondering if this a known issue (with a fix) or a new bug.  I would rather not re-install Java on 30 machines if its not necessary.

Thank you

**Comment by jr0cket** [comment]
With further investigation, the "No Java rutime present" occurs when you run nightcode from the GUI.

If I run `/Applications/Nightcode.app/Contents/MacOS/Nightcode` then the nightcode repl starts, reporting its using Clojure 1.9.0-alpha11

I set the JAVA_HOME variable in my `.profile` file, so GUI apps should detect its presence.

```
# Java runtime 
export JAVA_HOME='/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home'
export PATH=$PATH:JAVA_HOME/bin
```

Any suggestions for running it via the MacOSX GUI?  Thank you.

**Comment by oakes** [comment]
Hmm, not sure what's causing that. Was it happening in the previous version? I'm on the same version of OS X and I don't even have JAVA_HOME set on my machine, and I always launch it from the dock without issue.

**Comment by jr0cket** [comment]
I believe its the first time I have used Nightcode on this Mac.  I will try remove the JAVA_HOME variable from my profile and reboot (so the GUI environment forgets it).

I am also using zsh, but would be surprised if that is causing the issue.  I will see if anyone else has problems.  

The .deb file works just fine on my Ubuntu Linux laptop (my main machine).  I'll get some of the other ClojureBridge teachers with Macs to install Nightcode and see if they have any issue.

Thanks for your speedy response.

**Comment by jr0cket** [comment]
No one else seems to have this issue, so just some querk on my Mac setup.  I'll close the ticket as I use Linux for nearly everything.

Thanks for all the other help.

**Comment by oakes** [comment]
No problem! Just to note, it seems weird that the `JAVA_HOME` path would be in `/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/`. I figured it would be in `/Library/Java/JavaVirtualMachines/`.

**Comment by DoctorVin** [comment]
I'm having pretty much the identical issue with 2.1.8. I agree this is probably related to the environment on my mac, but I can't figure out what's up. 

`Lancer:~ drvince$ echo $JAVA_HOME
/Library/Java/JavaVirtualMachines/zulu8.72.jdk
Lancer:~ drvince$ java -version
openjdk version "1.8.0_72"
OpenJDK Runtime Environment (Zulu 8.13.0.5-macosx) (build 1.8.0_72-b15)
OpenJDK 64-Bit Server VM (Zulu 8.13.0.5-macosx) (build 25.72-b15, mixed mode)
Lancer:~ drvince$ javac -version
javac 1.8.0_72`

The one thing I did notice is that /usr/libexec/java_home returns nothing. But I have spent a while fruitlessly trying to get it to notice the JDK I have installed to no avail.

Any suggestions?
